src/Block/SharedBlockBlockService.php

```diff
@@ -130,20 +139,17 @@ public function load(BlockInterface $block): void
      */
     private function getBlockBuilder(AdminFormMapper $form): FormBuilderInterface
     {
-        $admin = $form->getAdmin();
```

Comment:
When you use the admin from the form, you are using BlockAdmin, but this field is to related with SharedBlockAdmin. What is more interesting is that this is happening inside PageAdmin, so for some reason, not only the admin linked is not the correct one which shows Blocks, not shared blocks. BUT also, for some reason Sonata is smart enough to know Block is child of Page and automatically builds child urls, showing only blocks related to that page. And the final funnier thing is that for child urls , the select button does not work for some reason I can't understand.

So to answer you:
1. No, the admin is not BlockAdmin but SharedBlockAdmin
2. Previously only blocks belonging to the page were shown (instead of the expected behavior which is showing ALL shared blocks)
3. Sonata has a bug somewhere (I didn't fully understand why) that make the previous implementation fail because it tries to open a modal with a child, but without providing the correct linkParameters (it is hard to explain because I do not fully know what is happening there)
4. But the bug pointed on 3. is not really related to this issue, which is: We were relating to block instead of shared block.

Maybe we want to investigate more about this 3. in the future.